packages/backend/src/logger/customLogger.ts

```diff
@@ -42,15 +50,38 @@ const transports = {
       ),
     }),
   ],
-  auditLog: [
-    new winston.transports.Console({
-      format: winston.format.combine(
-        auditLogFormat({ isAuditLog: true }),
-        defaultFormat,
-        winston.format.json(),
-      ),
-    }),
-  ],
+  auditLog: (config?: Config) => {
+    if (config?.getOptionalBoolean('logToConsole.enabled') === false) {
+      return [];
+    }
+    return [
+      new winston.transports.Console({
+        format: auditLogWinstonFormat,
+      }),
+    ];
+  },
+  auditLogFile: (config?: Config) => {
+    if (!config?.getOptionalBoolean('rotateFile.enabled')) {
+      return [];
+    }
+    return [
+      new winston.transports.DailyRotateFile({
+        format: auditLogWinstonFormat,
+        dirname:
+          config?.getOptionalString('rotateFile.logFileDirPath') ||
```

Comment:
```suggestion
          config?.getOptionalString('rotateFile.logFileDirPath') ??
```
I'm pretty sure SonarQube prefers the Nullish coalescing operator (??) over the Logical OR (||).

Reply:
oh I've gotten used to it since we use `||` over `??` in so much of our code when getting the values from the `Config`

Reply:
SonarQube complains on basically every instance of `||` 😭

Reply:
@Zaperex can you please fix this?

Reply:
oh I thought I already pushed the commit, but I guess I forgot. Should be fixed in https://github.com/janus-idp/backstage-showcase/pull/1443/commits/1f95a2f812d325762ab5ca1567e375d4bd57faf4